eyecite/find_citations.py

```diff
@@ -173,86 +176,48 @@ def extract_supra_citation(
     Supra 3: Adarand, supra, somethingelse
     Supra 4: Adrand, supra. somethingelse
     """
-    # Get page, with bounds check to ensure we don't scan past end of words.
-    # Use index + 2 to skip "at".
-    page = None
-    if index + 2 < len(words):
-        page = parse_page(str(words[index + 2]))
-
-    # Get antecedent -- either previous word, or previous two words if
-    # previous word is a comma. If previous word is a digit, store as
-    # volume and use next previous word as antecedent.
+    pin_cite, span_end = extract_pin_cite(words, index)
     antecedent_guess = None
     volume = None
-    if index > 0:
-        antecedent_guess = str(words[index - 1])
-        if antecedent_guess.isdigit():
-            volume = antecedent_guess
-            if index > 1:
-                antecedent_guess = str(words[index - 2])
-            else:
-                antecedent_guess = None
-        elif antecedent_guess == ",":
-            if index > 1:
-                antecedent_guess = str(words[index - 2]) + ","
-            else:
-                antecedent_guess = None
+    m = match_on_tokens(
+        words,
+        index - 1,
+        SUPRA_ANTECEDENT_REGEX,
+        strings_only=True,
+        forward=False,
+    )
+    if m:
```

Comment:
This is a little opaque, maybe add a short comment explaining that here we're checking if the antecedent could be a volume number instead of just a word? (This is clear in the regex definition in`helpers.py`, but not here.)

Reply:
Fixed with regex library.